features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -37,10 +37,11 @@ import java.util.concurrent.atomic.AtomicReference
 /**
  * Session Replay feature class, which needs to be registered with Datadog SDK instance.
  */
-internal class SessionReplayFeature constructor(
+internal class SessionReplayFeature(
     private val sdkCore: FeatureSdkCore,
     customEndpointUrl: String?,
     internal val privacy: SessionReplayPrivacy,
+    private val sampleRate: Float,
     private val rateBasedSampler: Sampler,
```

Comment:
it is kind of duplication of the same. normally we should be able to get sample rate from `rateBasedSampler.getSampleRate`

Reply:
yes but for that I need to cast the sampler to `RateBasedSampler` and I want to avoid that

Reply:
No, `getSampleRate` is a method of `Sampler` interface, no case is needed.

Reply:
oh ...didn't check that, good one ;)